src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -1039,6 +1068,124 @@ public ListenableFuture<ListResponse> list(String namespace, String prefix, Stri
+     *
+     *     ...
+     *
+     *     String uri = "s3://path/to/file.parquet";
+     *     String integrationId = "123-456-789";
+     *     StartImportResponse response = asyncIndex.startImport(uri, integrationId, ImportErrorMode.OnErrorEnum.CONTINUE);
+     *  }</pre>
+     *
+     * @param uri The URI prefix under which the data to import is available.
+     * @param integrationId The ID of the storage integration to access the data. Can be null or empty.
+     * @param errorMode Indicates how to respond to errors during the import process. Can be null.
+     * @return {@link StartImportResponse} containing the details of the initiated import operation.
+     * @throws ApiException if there are issues processing the request or communicating with the server.
+     *         This includes network issues, server errors, or serialization issues with the request or response.
+     */
+    public StartImportResponse startImport(String uri, String integrationId, ImportErrorMode.OnErrorEnum errorMode) throws ApiException {
+        StartImportRequest importRequest = new StartImportRequest();
+        importRequest.setUri(uri);
+        if(integrationId != null && !integrationId.isEmpty()) {
+            importRequest.setIntegrationId(integrationId);
+        }
+        if(errorMode != null) {
+            ImportErrorMode importErrorMode = new ImportErrorMode().onError(errorMode);
+            importRequest.setErrorMode(importErrorMode);
+        }
+
+        return bulkOperations.startBulkImport(importRequest);
+    }
+
+    /**
+     * <p>Lists all recent and ongoing import operations for the specified index.</p>
+     *
+     * <p>The method constructs a request to fetch a list of import operations, limited by the specified
+     * maximum number of operations to return per page. The pagination token allows for
+     * deterministic pagination through the list of import operations.</p>
+     *
+     * <p>Example:
+     *  <pre>{@code
+     *     import org.openapitools.db_data.client.ApiException;
+     *     import org.openapitools.db_data.client.model.ListImportsResponse;
+     *
+     *     ...
+     *     int limit = 10;
+     *     String paginationToken = "some-pagination-token";
+     *     ListImportsResponse response = asyncIndex.listImport(limit, paginationToken);
+     *  }</pre>
+     *
+     * @param limit The maximum number of operations to return per page. Default is 100.
+     * @param paginationToken The token to continue a previous listing operation. Can be null or empty.
+     * @return {@link ListImportsResponse} containing the list of recent and ongoing import operations.
+     * @throws ApiException if there are issues processing the request or communicating with the server.
+     *         This includes network issues, server errors, or serialization issues with the request or response.
+     */
+    public ListImportsResponse listImport(Integer limit, String paginationToken) throws ApiException {
```

Comment:
Should this be plural?

```suggestion
    public ListImportsResponse listImports(Integer limit, String paginationToken) throws ApiException {
```

Also, an overload of this method with no arguments could be useful since the limit is supposed to have default 100 and pagination token is only known for pages 2+ of the results.

Reply:
Fixed!